packages/dotcom-server-handlebars/src/helpers/renderJsxComponent.ts

```diff
@@ -0,0 +1,36 @@
+import path from 'path'
+import React from 'react'
+import ReactDOMServer from 'react-dom/server'
+
+export default function renderJsxComponent({ hash }) {
+  let modulePath
+
+  if (hash.hasOwnProperty('package')) {
+    modulePath = path.join(process.cwd(), `node_modules/${hash.package}`)
+  }
+
+  // localRoute is relative to root of app consuming dotcom-server-handlebars.
+  if (hash.hasOwnProperty('localRoute')) {
+    modulePath = path.resolve(process.cwd(), `./${hash.localRoute}`)
+  }
+
+  if (!modulePath) {
+    throw new Error('You must specify a "package" or "localRoute" argument to load a module')
+  }
+
+  const importedModule = require(modulePath)
+
+  const Component = hash.hasOwnProperty('namedExport')
+    ? importedModule[hash.namedExport]
+    : importedModule.default
+
+  const props = { ...this, ...hash }
+
+  if (props.hasOwnProperty('_locals')) {
```

Comment:
`this` will include `_locals`, meaning this error will always fire. Should we delete that property instead of throwing the error based on its presence?

Reply:
`this` should only include `_locals` at the root/0 depth level.

There is more info about it here: https://github.com/Financial-Times/x-dash/pull/396

Reply:
Yup, `this` does only include `_locals` at the root level but that is the thing that will always cause the error to throw. Think I have missed something.

Simplified, `this` is `{ _locals: {} }`

And `hash` is `{ foo: 'bar' }`

Meaning `const props = { ...this, ...hash }` leaves `props` as `{ _locals: {}, foo: 'bar' }`, meaning `props.hasOwnProperty('_locals')` will always evaluate to true, and always throw the error.

Are you saying we should include this check `this === data.root ? {} : this` that was removed in the referenced PR?

Reply:
Agreed offline that we should remove this check - done in [this commit](https://github.com/Financial-Times/dotcom-page-kit/pull/746/commits/8083fe47cdd7eb1490adb0c95c579185179a533e).